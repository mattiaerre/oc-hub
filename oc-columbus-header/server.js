export const data = (context, callback) => {
  const model = {
    logo: {
      href: '',
      url: context.params.logoUrl,
      alt: `${context.params.title} logo`,
    },
    title: context.params.title,
    timezoneOffset: '?'
  };

  if (context.plugins.getTimezoneOffset) {
    model.timezoneOffset = context.plugins.getTimezoneOffset();
  }

  callback(null, model);
};
