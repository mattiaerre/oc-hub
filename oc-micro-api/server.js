export const data = (context, callback) => {
  if (context.params.render === 'api') {
    callback(null, {
      message: 'Hello OC Î¼-API!'
    });
  } else {
    callback(null, {
      baseUrl: context.baseUrl,
      staticPath: context.staticPath
    });
  }
};
