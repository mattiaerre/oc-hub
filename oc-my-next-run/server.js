const React = require('react');
const { renderToString } = require('react-dom/server');
const { styleSheet } = require('styled-components');

const App = require('./src/App').default;
const runs = require('./package.json').runs;

export const data = (context, callback) => {
  const run = runs[context.params.id];
  const model = {
    twitterUsername: 'mattiaerre',
    when: new Date(...run['date-time']),
    where: {
      googleMapsShortUrl: run['google-maps-short-url']
    },
    note: run.note
  };
  const props = { model };
  const app = renderToString(<App {...props} />);
  const css = styleSheet.getCSS();
  callback(null, {
    app,
    model,
    css,
    staticPath: context.staticPath
  });
};
