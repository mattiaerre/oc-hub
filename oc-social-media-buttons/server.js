const React = require('react');
const { renderToString } = require('react-dom/server');
const { styleSheet } = require('styled-components');
const App = require('./src/App').default;

export const data = (context, callback) => {
  const app = renderToString(<App />);
  const css = styleSheet.getCSS();
  callback(null, { app, css });
};
