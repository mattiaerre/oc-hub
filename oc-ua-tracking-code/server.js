export const data = (context, callback) => {
  callback(null, { trackingId: context.params.trackingId });
};
