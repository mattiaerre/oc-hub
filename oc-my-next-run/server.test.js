import { data } from './server';

test('viewModel to match snapshot', () => {
  const context = { params: { id: '2017-03-19' }, staticPath: '/' };
  data(context, (error, viewModel) => {
    expect(viewModel).toMatchSnapshot();
  });
});
