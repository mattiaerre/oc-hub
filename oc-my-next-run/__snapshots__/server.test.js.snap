// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewModel to match snapshot 1`] = `
Object {
  "app": "<section class=\\"sc-bwzfXH bOYEsa\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-676159848\\"><h1 class=\\"sc-bdVaJa kbYEP\\" data-reactid=\\"2\\">@mattiaerre&#x27;s next run</h1><p data-reactid=\\"3\\">March 19th 2017, 9:15:00 am</p><p data-reactid=\\"4\\"><a href=\\"https://goo.gl/maps/3i6YF36SEPH2\\" data-reactid=\\"5\\">https://goo.gl/maps/3i6YF36SEPH2</a></p><ul data-reactid=\\"6\\"><li data-reactid=\\"7\\">Run, easy pace, 165 minutes.</li><li data-reactid=\\"8\\">Cool down, 5 to 10 minutes.</li><li data-reactid=\\"9\\">Stretch.</li></ul></section>",
  "css": ".sc-bdVaJa {}.kbYEP {font-size: 1.5em;text-align: center;}.sc-bwzfXH {}.bOYEsa {padding: 2em;background: floralwhite;font-family: monospace;border: 1px solid grey;}",
  "model": Object {
    "note": Array [
      "Run, easy pace, 165 minutes.",
      "Cool down, 5 to 10 minutes.",
      "Stretch.",
    ],
    "twitterUsername": "mattiaerre",
    "when": 2017-03-19T16:15:00.000Z,
    "where": Object {
      "googleMapsShortUrl": "https://goo.gl/maps/3i6YF36SEPH2",
    },
  },
  "staticPath": "/",
}
`;
